packages/node-core/src/indexer/worker/worker.connectionPoolState.manager.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {Injectable} from '@nestjs/common';
+import {ApiConnectionError, ApiErrorType} from '../connectionPool.service';
+
+export type HostConnectionPoolState<T> = {
+  hostGetNextConnectedApiIndex: () => Promise<number | undefined>;
+  hostAddToConnections: (endpoint: string, index: number) => Promise<void>;
+  hostGetFieldFromConnectionPoolItem: <K extends keyof T>(index: number, field: K) => Promise<T[K]>;
+  hostSetFieldInConnectionPoolItem: <K extends keyof T>(index: number, field: K, value: T[K]) => Promise<void>;
+  hostSetTimeoutIdInConnectionPoolItem: (index: number, delay: number) => Promise<void>;
+  hostClearTimeoutIdInConnectionPoolItem: (index: number) => Promise<void>;
+  hostGetSuspendedIndices: () => Promise<number[]>;
+  hostDeleteFromPool: (index: number) => Promise<void>;
+  hostHandleApiError: (index: number, errorType: string) => Promise<void>;
+  hostHandleApiSuccess: (index: number, responseTime: number) => Promise<void>;
+  hostGetDisconnectedIndices: () => Promise<number[]>;
+  hostShutdownPoolState: () => Promise<void>;
+};
+
+export const hostConnectionPoolStateKeys: (keyof HostConnectionPoolState<any>)[] = [
+  'hostGetNextConnectedApiIndex',
+  'hostAddToConnections',
+  'hostGetFieldFromConnectionPoolItem',
+  'hostSetFieldInConnectionPoolItem',
+  'hostSetTimeoutIdInConnectionPoolItem',
+  'hostClearTimeoutIdInConnectionPoolItem',
+  'hostGetSuspendedIndices',
+  'hostDeleteFromPool',
+  'hostHandleApiError',
+  'hostHandleApiSuccess',
+  'hostGetDisconnectedIndices',
+  'hostShutdownPoolState',
+];
+
+@Injectable()
+export class WorkerConnectionPoolStateManager<T> {
```

Comment:
This should implement an interface so we know that it matches the non-worker version